src/app/components/wrapper-components/part-creation/part-creation.component.spec.ts

```diff
@@ -9,620 +12,719 @@ import { PartDAO } from 'src/app/dao/PartDAO';
+                await userDAO.update(UserMocks.CREATOR_AUTH_USER.id, { state: 'offline' });
+                await partDAO.set('joinerId', PartMocks.INITIAL);
 
                 // When arriving on that component
-                testUtils.detectChanges();
-                tick(3000); // needs to be >2999
+                tick();
 
                 // Then the part and all its related data should be removed
-                expect(gameService.deletePart).toHaveBeenCalledWith('joinerId');
-                expect(joinerService.deleteJoiner).toHaveBeenCalledWith();
-                expect(chatService.deleteChat).toHaveBeenCalledWith('joinerId');
+                expect(gameService.deletePart).not.toHaveBeenCalled();
+                expect(joinerService.deleteJoiner).not.toHaveBeenCalled();
+                expect(chatService.deleteChat).not.toHaveBeenCalled();
+                component.stopSendingPresenceTokensAndObservingUsersIfNeeded();
+            }));
+        });
+        describe('Leaving', () => {
+            it('should remove yourself when leaving the room and empty user.observedPart', fakeAsync(() => {
+                // Given a partCreation where user is candidate
+                awaitComponentInitialisation();
+                expect(component.currentJoiner).toEqual(JoinerMocks.WITH_FIRST_CANDIDATE);
+
+                // When leaving the page (tested here by calling ngOnDestroy)
+                const authService: ConnectedUserService = TestBed.inject(ConnectedUserService);
+                spyOn(authService, 'removeObservedPart').and.callThrough();
+                spyOn(joinerService, 'unsubscribe').and.callFake(() => {});
+                spyOn(joinerService, 'cancelJoining').and.callFake(async(user: MinimalUser) => {});
+                void component.ngOnDestroy();
+                tick();
+
+                // Then joinerService.cancelJoining should have been called
+                expect(joinerService.cancelJoining).toHaveBeenCalledOnceWith(UserMocks.OPPONENT_MINIMAL_USER);
+                expect(authService.removeObservedPart).toHaveBeenCalledOnceWith();
             }));
         });
-        describe('Waiting for config', () => {
-            it('should reroute to server when game is cancelled', fakeAsync(async() => {
+        describe('Hoping to get chosen', () => {
+            it('should reroute to server when game is cancelled', fakeAsync(() => {
                 const router: Router = TestBed.inject(Router);
                 spyOn(router, 'navigate');
 
                 // Given component that has loaded
-                testUtils.detectChanges();
-                await testUtils.whenStable();
+                awaitComponentInitialisation();
 
                 // When the joiner is deleted (because the game has been cancelled)
-                await joinerDAOMock.delete('joinerId');
-
-                // Then the user is rerouted to the server
+                void joinerDAO.delete('joinerId');
```

Comment:
await instead of void? T'sais voir ceux qui marchent pour  limiter le nombre de void au max ?